### PR TITLE
Support CAST from Decimal datatype to String

### LINKED ISSUE #3991: Support Decimals cast to Utf8/LargeUtf
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Currently the CAST is not supported from Decimal to UTF. This ticket to add the necessary cast

**Describe the solution you'd like**

Support Decimals cast to Utf8/LargeUtf

**Describe alternatives you've considered**

None
**Additional context**

None

### PR DESCRIPTION
**Pull request by comphead**
# Which issue does this PR close?

Closes #3991.

# Rationale for this change
 Support cast Decimal datatype to string

# What changes are included in this PR?

# Are there any user-facing changes?
No

### CODE REVIEW

**Comment by comphead** [comment]
@tustvold not sure why the test fails on         
```
Test casting Decimal128(38, 0) --> LargeList(Field { name: "item", data_type: Utf8, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} })
thread 'test_can_cast_types' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidArgumentError("non-nullable child of type Utf8 contains nulls not present in parent LargeList(Field { name: \"item\", data_type: Utf8, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} })")', arrow-data/src/data/mod.rs:1795:30
```

Perhaps related to #3983

**Comment by tustvold** [comment]
My guess is the cast is failing and inserting a null, but the field doesn't permit nulls. I'm not sure what the cast kernel should do in such a case tbh, but in the immediate term I'd suggest finding a way to avoid the decimal cast from failing

**Comment by comphead** [comment]
> My guess is the cast is failing and inserting a null, but the field doesn't permit nulls. I'm not sure what the cast kernel should do in such a case tbh, but in the immediate term I'd suggest finding a way to avoid the decimal cast from failing

@tustvold I have removed `None` from decimal array as other tests casts with not nullable values

File: arrow-cast/src/cast.rs [hunk 1]
```diff
@@ -8132,4 +8139,68 @@ mod tests {
...
+        ) {
+            assert!(can_cast_types(&input_type, &output_type));
+
+            let b = cast(&array, &output_type).unwrap();
+
+            assert_eq!(b.data_type(), &output_type);
+            let c = b
+                .as_any()
+                .downcast_ref::<GenericByteArray<GenericStringType<OffsetSize>>>()
+                .unwrap();
```
Comment (tustvold) [inline arrow-cast/src/cast.rs:8158]:
```suggestion
            let c = b.as_string::<OffsetSize>()
```

File: arrow-cast/src/cast.rs [hunk 2]
```diff
@@ -8132,4 +8139,68 @@ mod tests {
             .unwrap();
         assert_eq!(1672531200000000000, c.value(0));
     }
+
+    #[test]
+    fn test_cast_decimal_to_utf8() {
+        fn test_decimal_to_string<IN: ArrowPrimitiveType, OffsetSize: OffsetSizeTrait>(
```
Comment (tustvold) [inline arrow-cast/src/cast.rs:8145]:
```suggestion
        fn test_decimal_to_string<OffsetSize: OffsetSizeTrait>(
```

File: arrow-cast/src/cast.rs [hunk 3]
```diff
@@ -8132,4 +8139,68 @@ mod tests {
...
+            input_type: DataType,
```
Comment (tustvold) [inline arrow-cast/src/cast.rs:8146]:
```suggestion
```

**Review by tustvold — APPROVED** [review]
Just some minor nits

**Comment by tustvold** [comment]
I took the liberty of resolving the merge conflicts on this

**Pull request merged**